test/test_iscsi.py

```diff
@@ -0,0 +1,327 @@
+import unittest
+from unittest.mock import Mock, patch, MagicMock
+import ddt
+from cinder.volume.drivers.ixsystems.common import TrueNASCommon
+from cinder.volume.drivers.ixsystems.freenasapi import FreeNASServer
+from cinder.volume.drivers.ixsystems.iscsi import FreeNASISCSIDriver
+from cinder.volume import configuration as conf
+from cinder.tests.unit import fake_volume
+from cinder import context
+
+class fakecommon(TrueNASCommon):
+
+    def __init__(self, configuration=None):
+        CONF = MagicMock(spec=conf.Configuration)
+        CONF.iscsi_helper = 'tgtadm'
+        CONF.volume_dd_blocksize = 512
+        CONF.volume_driver = 'cinder.volume.drivers.ixsystems.iscsi.FreeNASISCSIDriver'
+        CONF.ixsystems_login = 'root'
+        CONF.ixsystems_password = 'Pa55w0rd'
+        CONF.ixsystems_apikey = ''
+        CONF.ixsystems_server_hostname = '10.3.1.81'
```

Comment:
why are these hard-coded?

Reply:
Okey, I did update and extracting the config into fake_config_dict dictionary object, make it easier to maintain.
https://github.com/YiHuangIX/cindertruenas/blob/upstream/test/test_common.py#L309

PS: I checked most of the existing cinder driver test script, all of them are hard coded without defending external configuration resource file. 
https://github.com/openstack/cinder/blob/master/cinder/tests/unit/volume/drivers/ceph/test_rbd_iscsi.py#L48

There reasons behand I assume might those scripts need to be self encapsulate and used for sanity test purpose instead of full coverage of unit test script.